CMakeLists.txt

```diff
@@ -410,13 +411,58 @@ else()
     set(DFHACK_TINYXML "dfhack-tinyxml")
 endif()
 
+find_package(ZLIB REQUIRED)
+
+if(BUILD_CORE_TESTS)
+    include_directories(depends/googletest/googletest/include)
```

Comment:
this should be set as a property for test files (similar to how the COMPILE_FLAGS are set on line 438). it shouldn't be included in the header search directory for library code, right?

Reply:
having it here allows anybody to start writing unit tests plugins and library alike

Reply:
Yeah, adding include paths based on cmake options is likely to cause build errors if e.g. someone introduces code that depends on the added include path. That effectively makes the cmake option required to build, at which point it isn't an option at all.

Enabling _targets_ based on cmake options is fine, and something we do already. So I think tests should be gated by an option, and the include path should be enabled on just those targets.

Reply:
> so the tests are intended to be written in plugin and core code? not just test files? I don't agree with that plan. I think test code should be restricted to test files, like the example test in this PR.

no?? It means they can write unit tests for plugins and library code alike.. I don't think it's an easy feat to simply write unit tests in the code being tested with the whole: it needs to have a `main()` to function as a unit test.

The point is, that library tests will be with library code and plugin tests would be with the plugin code. Two different sub-directories that would otherwise need to be provided the include path separately.

Reply:
> Enabling _targets_ based on cmake options is fine, and something we do already. So I think tests should be gated by an option, and the include path should be enabled on just those targets.

I can look into it